Tests/Common/SimulatedDevice.cs

```diff
@@ -129,15 +125,27 @@ public LoRaPayloadData CreateUnconfirmedDataUpMessage(string data, uint? fcnt =
                 direction,
                 Supports32BitFCnt ? fcnt : null);
 
+            if (AppSKey is null || NwkSKey is null) throw new InvalidOperationException("Can't perform encryption as some keys are empty");
+            payloadData.PerformEncryption(string.IsNullOrEmpty(appSKey) ? AppSKey : appSKey);
+            payloadData.SetMic(string.IsNullOrEmpty(nwkSKey) ? NwkSKey : nwkSKey);
```

Comment:
Are you sure you don't mean the following?

```suggestion
            payloadData.PerformEncryption(appSKey is { Length: > 0 } ? appSKey : AppSKey ?? throw new InvalidOperationException($"Can't perform encryption without {nameof(AppSKey)}."));
            payloadData.SetMic(nwkSKey is { Length: > 0 } ? nwkSKey : NwkSKey ?? throw new InvalidOperationException($"Can't perform encryption without {nameof(NwkSKey)}."));
```